pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -72,13 +75,27 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 
 	sourceCloudConfigMap := infrastructure.Spec.CloudConfig.Name
 	sourceCloudConfigNamespace := configNamespace
+	sourceCloudConfigKey := infrastructure.Spec.CloudConfig.Key
+
+	// If a managed cloud-provider config is available, it should be used instead of the default.
+	_, err = listers.ConfigMapLister().ConfigMaps(machineSpecifiedConfigNamespace).Get(machineSpecifiedConfig)
+	if err != nil {
```

Comment:
```go
if !errors.IsNotFound(err) {
```
is sufficient.

Reply:
@soltysh thank you for the review. I think using your suggestion with my original code would omit the case where `err == nil && err.IsNotFound(err)` in which case we want to keep the default values. So I think my code was unclear and I rewrote this conditional. PTAL and let me know if there are any further changes.

Reply:
Yeah, this is much better now. Thx!